@FIR-204: Revised and sanitized changes from SSD

atrivedi-tsavoritesi opened the pull request:
Removed following changes from patch
1. Enable PCI-E root port in DTS
2. Remove forcing of Gen 2 in DTS
3. Remove forcing of 4 io queues
4. Remove increased timeouts in nvme code

@mmankal I have only kept code which is absolutely needed. We can look at config flags as well but right now all of the above changes are not needed in the code.

Commit:
@FIR-204: Remove not needed changes from SSD
Removed following changes from patch
1. Enable PCI-E root port in DTS
2. Remove forcing of Gen 2 in DTS
3. Remove forcing of 4 io queues
4. Remove increased timeouts in nvme code

mmankal reviewed (commented):
I wanted to multiple optimizations.  I  want to try out these and  until then please do not merge this code to main branch

akapoor3518 reviewed (approved):
lgtm

Commit:
@FIR-204: Removed the config flags which are not needed
Removed config flags which are not needed
The NVME Multipath and the NVME Debug Errors are not needed

mmankal reviewed (commented):
All the code that was in these files were necessary changes that were needed for troubleshooting and debug of the NVME driver for Sandisk 1TB M.2 . With this code the driver worked just fine. However, in the interest of reducing the code footprint and to keeping the code as close as possble to the main linux-socfpga - some of these chnages were revised and sanitized.
